### PR TITLE
Implement deleting search documents

### PR DESCRIPTION
**Pull request by Toflar**

### CODE REVIEW

File: core-bundle/src/Search/Backend/BackendSearch.php [hunk 1]
```diff
@@ -143,8 +176,11 @@ private function convertSearchDocumentToProviderHit(array $document): Hit|null
         $document = Document::fromArray(json_decode($document['document'], true, 512, JSON_THROW_ON_ERROR));
         $hit = $fileProvider->convertDocumentToHit($document);
 
+        // The provider did not find any hit for it anymore so it must have been removed
+        // or expired. Remove from the index.
         if (!$hit) {
-            // TODO: delete this document from the index
+            $this->deleteDocuments([$hit->getDocument()]);
```
Comment (leofeyer) [inline core-bundle/src/Search/Backend/BackendSearch.php]: This cannot work because `if (!$hit)`.

Comment (Toflar) [reply core-bundle/src/Search/Backend/BackendSearch.php]: Yes I saw that. Will write a unit test that covers this case.

Comment (Toflar) [reply core-bundle/src/Search/Backend/BackendSearch.php]: Fixed.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**